lib/main.dart

```diff
@@ -18,22 +32,40 @@ class MyApp extends StatefulWidget {
 }
 
 class _MyAppState extends State<MyApp> {
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MultiBlocProvider(
       providers: [BlocProvider(create: (context) => WorldClockBloc())],
-     child:
-        MaterialApp(
-            title: 'World_Clock',
-            theme: ThemeData(
-              primarySwatch: Colors.blue,
-            ),
-            routes: {
-              Screens.clockScreen: (context) => ClockScreen(),
-              Screens.addTimeScreen: (context) => AddTimeScreen()
-            })
-
+      child: Builder(builder: (context) {
+        Hive.openBox(clocksKeeperKey).then((box) {
+          if (box.isNotEmpty) {
+            List<WorldClockModel> clockList =
+                box.get(clocksModelKey).cast<WorldClockBloc>();
+            if (clockList.isEmpty) {
+              context.read<WorldClockBloc>().add(EmptyListEvent());
+            } else {
+              for (int index = 0; index < clockList.length; index++) {
+                context
+                    .read<WorldClockBloc>()
+                    .add(SaveClockEvent(clockList[index]));
```

Comment:
it's better to have SaveAll event type in such cases
but check my other comments on this file first as that part shouldn’t actually be here